src/amqp-ts.ts

```diff
@@ -166,44 +167,55 @@ export class Connection extends EventEmitter {
   }
 
   _rebuildAll(err: Error): Promise<void> {
-    log.log("warn", "Connection error: " + err.message, { module: "amqp-ts" });
+    if (this.rebuildPromise) {
+      return this.rebuildPromise;
```

Comment:
Does that mean without making `_rebuildAll` idempotent, we are rebuilding many connections?

Then each connection will create x channels for queues/exchanges on initialize():
https://github.com/zendesk/amqp-ts/blob/44677802318d288f634a5c52ec00f16617b6e026/src/amqp-ts.ts#L812
https://github.com/zendesk/amqp-ts/blob/44677802318d288f634a5c52ec00f16617b6e026/src/amqp-ts.ts#L493

Reply:
It's a single connection, but it rebuilds its channel list many times. So that's how the number spirals out of control and we hit the max

Reply:
Yea, we re-read the code rebuilding connection is fine even if we call it multiple times but it's the steps after on exchanges/queues that probably rebuild that many channels. 😢